pkg/provider/helm/helm_test.go

```diff
@@ -22,3 +23,94 @@ func TestHelmSPIREProvider(t *testing.T) {
 	assert.Equal(t, p.SPIRECRDsVersion, "0.4.0")
 	assert.Equal(t, trustZoneProto.TrustDomain, p.trustZone.TrustDomain)
 }
+
+func TestGetChartRef(t *testing.T) {
+	originalPath := os.Getenv("HELM_REPO_PATH")
+	defer os.Setenv("HELM_REPO_PATH", originalPath)
+
+	tests := []struct {
+		name            string
+		helmRepoPath    string
+		helmRepoPathSet bool
+		chartName       string
+		want            string
+		wantErr         bool
+		errString       string
+	}{
+		{
+			name:            "with HELM_REPO_PATH set",
+			helmRepoPathSet: true,
+			helmRepoPath:    "spire-local",
+			chartName:       "spire",
+			want:            "spire-local/spire",
+			wantErr:         false,
+		},
+		{
+			name:            "with HELM_REPO_PATH containing trailing slash",
+			helmRepoPathSet: true,
+			helmRepoPath:    "custom-repo/",
+			chartName:       "spire",
+			want:            "custom-repo/spire",
+			wantErr:         false,
+		},
+		{
+			name:            "with HELM_REPO_PATH containing trailing slashes",
+			helmRepoPathSet: true,
+			helmRepoPath:    "custom-repo//",
+			chartName:       "spire",
+			want:            "custom-repo/spire",
+			wantErr:         false,
+		},
+		{
+			name:            "with empty HELM_REPO_PATH",
+			helmRepoPathSet: true,
+			helmRepoPath:    "",
+			chartName:       "spire",
+			want:            "spire/spire",
+			wantErr:         true,
+			errString:       "HELM_REPO_PATH environment variable is set but empty",
+		},
+		{
+			name:            "with empty chart name",
+			helmRepoPathSet: false,
+			helmRepoPath:    "spire-local",
+			chartName:       "",
+			wantErr:         true,
+			errString:       "chart name cannot be empty",
+		},
+		{
+			name:            "with HELM_REPO_PATH not set",
+			helmRepoPathSet: false,
+			chartName:       "spire",
+			want:            "spire/spire",
+			wantErr:         false,
+		},
+		{
+			name:            "with empty HELM_REPO_PATH and an empty chart name",
+			helmRepoPathSet: false,
+			chartName:       "",
+			wantErr:         true,
+			errString:       "chart name cannot be empty",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.helmRepoPathSet {
+				os.Setenv("HELM_REPO_PATH", tt.helmRepoPath)
+			} else {
+				os.Unsetenv("HELM_REPO_PATH")
+			}
+
+			got, err := getChartRef(tt.chartName)
+			if tt.wantErr {
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), tt.errString)
```

Comment:
You can also do `assert.ErrorContains(t, err, tt.errString)` which saves about 3 characters :D

Reply:
Sounds good, I'll try that on the next one!